models/experimental.py

```diff
@@ -95,6 +95,13 @@ def attempt_load(weights, map_location=None, inplace=True, fuse=True):
     for w in weights if isinstance(weights, list) else [weights]:
         ckpt = torch.load(attempt_download(w), map_location=map_location)  # load
         ckpt = (ckpt.get('ema') or ckpt['model']).float()  # FP32 model
+
+        # backwards compatability for sparsezoo models that maintain anchor grid in state dict
+        anchor_grid_key = [key for key in ckpt.state_dict().keys() if "anchor_grid" in key]
+        if len(anchor_grid_key) == 1:
+            del ckpt.model[-1].anchor_grid
+            ckpt.model[-1].anchor_grid = [torch.zeros(1)] * ckpt.model[-1].nl # reset to init value
```

Comment:
So in this file you aren't actually removing it from state_dict, but replacing? Whereas other file you are just removing from state dict? How come they are different?

Reply:
Oh i see in the PR description that export just doesn't need anchor_grid? Is this experimental.py script part of the training stuff that needs anchor grid in the new format?

Reply:
This fix is highlighting the issue/complexity that comes with having 2+ load flows. Yolov5 typically saves the full model in pickled format rather than just the state dict. We had to add a state dict path because pickling isn't supported for quantized models. So now we have a path for each. 

**Load state dict fix**
- Model is initialized from scratch (and thus has the correct init anchor_grid) _[usual behavior]_
- `anchor_grid` is removed from the state dict _[added with fix]_
- State dict is loaded into model _[usual behavior]_

**Load pickled model fix**
- Pickled file is loaded _[usual behavior]_
- `anchor_grid` field is re-initialized _[added with fix]_
- Model is as the actual model, no state dict application _[usual behavior]_

Reply:
To clarify, for non-quantized models we still use the original(ish) path

Reply:
I see yeah that is complicated. thanks for the details!